rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/rename/Common.rsc

```diff
@@ -171,39 +194,116 @@ tuple[bool, bool](Tree) allNameSortsFilter(str name1, str name2) {
     };
 }
 
-set[loc] filterFiles(set[loc] fs, bool(Tree) treeFilter, Tree(loc) getTree) {
-    j = "Checking files for occurrences of relevant name";
+bool(Tree) anyNameFilter(set[str] names) {
+    set[str] escNamesS = {normalizeEscaping(name) | name <- names};
+    set[Name] escNames = {*escapeSet(#Name, name) | name <- escNamesS};
+    set[Nonterminal] escNonterminals = {*escapeSet(#Nonterminal, name) | name <- escNamesS};
+    set[NonterminalLabel] escNonterminalLabels = {*escapeSet(#NonterminalLabel, name) | name <- escNamesS};
+    set[QualifiedName] qualifiedNames = {t | name <- escNamesS, just(QualifiedName t) := tryParseAs(#QualifiedName, name)};
+
+    return bool(Tree tr) {
+        visit (tr) {
+            case (Name) `<Name n>`: for (n1 <- escNames, n := n1) return true;
+            case (Nonterminal) `<Nonterminal n>`: for (nt1 <- escNonterminals, n := nt1) return true;
+            case (NonterminalLabel) `<NonterminalLabel n>`: for (ntl1 <- escNonterminalLabels, n := ntl1) return true;
+            case (QualifiedName) `<QualifiedName n>`: {
+                if (n.names[0].src == n.src) { // skip unqualified names
+                    for (qn <- qualifiedNames, qn := n || qn := [QualifiedName] normalizeEscaping("<n>")) return true;
+                }
+            }
+        }
+
+        return false;
+    };
+}
+
+bool(loc) containsQualifiedNameCheck(str _) =
```

Comment:
Why does this one always return true? (Add a comment on the reasoning, like fallback to parse tree or something)

Reply:
Unused, removed